etw/cs/traceloggingdynamic/TraceLoggingDynamic.cs

```diff
@@ -674,13 +678,40 @@ private enum InType : byte
             Struct, // Use OutType for field count.
             CountedBinary, // counted, works for arrays, newer (not always supported)
             Mask = 31,
+            NullArray = 64, // Invalid type

```

Comment:
Note: This is just to make the syntax nicer. Previously we had two overloads of AddMetadata - one for array and one for scalar. The only difference between them was that the AddMetadataArray method OR'ed a flag into the intype value. As more code moved into these functions, it has become more obvious that they needed to merge, but the syntax would have been very ugly to do the OR at the call site. Fix is to create pre-OR'ed enumerators.